core/localnetwork.go

```diff
@@ -0,0 +1,59 @@
+package core
```

Comment:
The current implementation of `VideoCache` (`BasicVideoCache`) has a tight integration with basicnet.  I had the assumption that `VideoCache` implicitly has a data source, so it only need to have `Get` and `Evict` functions, and never need to have `Update` functions (`Get` will automatically check from the data source).

What about expanding the `VideoCache` interface to add `Update` functions?  We can rename `BasicVideoCache` to `BasicnetVideoCache`, and it will implement something like:
```
func UpdateMasterPlaylist(manifestID string, mpl *m3u8.MasterPlaylist) error {
  return errors.New("Cannot Update in BasicnetVideoCache")
}
```
and move `LocalNetwork` to `LocalVideoCache`, which can implement the `Evict` and `Update` functions.

Reply:
We can add `Update` functions to the interface, however I'm not really sure if we need a formalized interface so much anymore now that basicnet is gone. We'll see how things look after the VideoNetwork and the VideoCache/Source are combined.

However, at some point we will likely need to parameterize the VideoSource by the object store. That part can wait though.